crates/types/src/epoch_table.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(dead_code)]
+
+use crate::identifiers::{LeaderEpoch, PartitionId};
+use crate::{GenerationalNodeId, Version, Versioned};
+use std::collections::HashMap;
+
+/// Partition processor epoch table.
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct EpochTable {
+    version: Version,
+    epochs: HashMap<PartitionId, EpochMetadata>,
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Completely unrelated to this PR: I think we should stop gating "serde" it doesn't really make a lot of sense to continue with this noise. It definitely makes sense to gate things like `schemars` since they're only needed in specific tooling, but serde should be always on.

Gating serde would make sense if we were publishing a library, but in practical terms, every time you build almost anything on restate, you'll have to have serde enabled.

Reply:
Agreed.